diff --git a/networking/udhcp/d6_socket.c b/networking/udhcp/d6_socket.c
index acf1083..0110e60 100644
--- a/networking/udhcp/d6_socket.c
+++ b/networking/udhcp/d6_socket.c
@@ -74,7 +74,7 @@ int FAST_FUNC d6_read_interface(
 	freeifaddrs(ifap);
 
 	if (retval & (1<<0)) {
-		/* This iface has no MAC (e.g. ppp), generate a random one */
+		/* This iface has no MAC (e.g. ppp), set a static one */
 		struct ifreq ifr;
 		int fd;
 
@@ -86,9 +86,12 @@ int FAST_FUNC d6_read_interface(
 			log2("ifindex %d", *ifindex);
 			if (((uint32_t*)mac)[0] == 0) {
 				/* invent a fictitious MAC (once) */
-				((uint32_t*)mac)[0] = rand();
-				((uint16_t*)mac)[2] = rand();
-				mac[0] &= 0xfc; /* make sure it's not bcast */
+				mac[0] = 0x10;
+				mac[1] = 0x00;
+				mac[2] = 0x00;
+				mac[3] = 0x00;
+				mac[4] = 0x00;
+				mac[5] = 0x00;
 			}
 			retval &= (3 - (1<<0));
 		}
